src/App.js

```diff
@@ -11,6 +14,22 @@ import {  useState } from "react";
  */
 function App() {
   const [selectedQuery, setSelectedQuery] = useState(undefined)
+  const session = getDefaultSession()
+  const [loggedIn, setLoggedIn] = useState()
+  useEffect(() => {
+    session.onLogin(() => setLoggedIn(true))
+    session.onLogout(() => setLoggedIn(false))
+
+    handleIncomingRedirect({restorePreviousSession: true})
```

Comment:
Use async/await instead of then.

Reply:
This would need the useEffect argument function to be async, which according to the sources on the Web should be avoided and also causes an ESLint warning.

Reply:
Can you add this in the comments together with the sources?

Reply:
Do you mean comments of this PR?

Reply:
No, in the code itself.

Reply:
I have done this [commit 233fbe9](https://github.com/SolidLabResearch/generic-data-viewer/pull/23/commits/233fbe9a0ca06ad4fe3c058358c3e413f2303fc8)